backend/file-ucloud-service/src/jvmMain/kotlin/services/SyncthingClient.kt

```diff
@@ -443,28 +443,316 @@ class SyncthingClient(
+                                                    it.getField(SyncFoldersTable.id) == row.getField(
+                                                    SyncFoldersTable.id
+                                                ) &&
+                                                    it.getField(SyncDevicesTable.user) == row.getField(
+                                                    SyncFoldersTable.user
+                                                )
+                                            }.map {
+                                                SyncthingFolderDevice(it.getField(SyncDevicesTable.device))
+                                            },
+                                        path = File(
+                                            "/mnt/sync",
+                                            row.getField(SyncFoldersTable.id).toString()
+                                        ).absolutePath,
+                                        type = SynchronizationType.valueOf(row.getField(SyncFoldersTable.syncType)).syncthingValue,
+                                        rescanIntervalS = device.rescanIntervalSeconds
+                                    )
+                                }
+
+                            try {
+                                val resp = httpClient.put<HttpResponse>(deviceEndpoint(device, "/rest/config/folders")) {
+                                    body = TextContent(
+                                        defaultMapper.encodeToString(newFolders),
+                                        ContentType.Application.Json
+                                    )
                                     headers {
                                         append("X-API-Key", device.apiKey)
                                     }
                                 }
 
-                                if (resp.status == HttpStatusCode.OK) {
+                                if (resp.status != HttpStatusCode.OK) {
+                                    throw RPCException(
+                                        resp.content.toByteArray().toString(Charsets.UTF_8),
+                                        dk.sdu.cloud.calls.HttpStatusCode.BadRequest
+                                    )
+                                } else {
                                     pendingDevices = pendingDevices.filter { it.id != device.id }
                                 }
+
+                            } catch (ex: RPCException) {
+                                throw RPCException("Invalid Syncthing Configuration", dk.sdu.cloud.calls.HttpStatusCode.BadRequest)
+                            } catch (ex: Throwable) {
+                                // Do nothing
+                            }
+                        }
+                    }
+                    lastWrite.set(Time.now())
+
+                    if (Time.now() > timeout) {
```

Comment:
I am not entirely sure what the timeout does. As far as I can tell, we wait for the request to finish regardless. So throwing at this point, when it has finally completed, seems wasteful.